src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserHelperClasses.cs

```diff
@@ -329,6 +341,59 @@ internal string tableName
             }
         }
 
+
+        public byte Updatability
+        {
+            get => (byte)(flags & _SqlMetadataFlags.IsReadOnlyMask);
+            set => flags = (_SqlMetadataFlags)((value & 0x3) | ((int)flags & ~0x03));
+        }
+
+        public bool IsReadOnly
+        {
+            get => (flags & _SqlMetadataFlags.IsReadOnlyMask ) == _SqlMetadataFlags.None;
+        }
+
+        public bool IsDifferentName
+        {
+            get => (flags & _SqlMetadataFlags.IsDifferentName) == _SqlMetadataFlags.IsDifferentName;
```

Comment:
Can be `get => flags.HasFlag(_SqlMetadataFlags.IsDifferentName);`

Reply:
Using Enum will cause the parameter to box, does reflection and null checks and causes a new stackframe. Since it's internal i know there can't be rogue data in there so it's faster to mask.

Reply:
After discussing with some folks on gitter i've changed my mind on this and changed the code. In coreclr the HasFlags implementation is intrinsic and will have good perf, on netstandard the perf won't be great bit it's not a on a hot cpu path.